src/main/java/pl/umk/mat/git2befit/service/workout/factory/implementation/SplitTrainingPlan.java

```diff
@@ -17,68 +19,71 @@
     private final List<String> smallBodyParts = List.of("SIXPACK", "CALVES", "BICEPS", "TRICEPS", "SHOULDERS");
     private final List<String> bigBodyParts = List.of("CHEST", "BACK", "THIGHS");
 
-    private final ExerciseRepository exerciseRepository;
-
+    private List<Exercise> exercisesWithoutEquipment = new ArrayList<>();
+    private List<Exercise> exercisesWithEquipment = new ArrayList<>();
     private TrainingForm trainingForm;
+    private TrainingForm localTrainingForm;
```

Comment:
Wydaje mi się, że powinniśmy serio pomyśleć o wywaleniu ```@Component``` albo zrobić, żeby to posiadało Scope na request, bo nie wiem czy tak przechowując rzeczy nie może wyjść jakieś gówno. Z drugiej strony faktorka zawsze tworzy nowy obiekt, to może ten pierwszy pomysł jest najsensowniejszy.

Reply:
To że jest @Component to tak naprawdę nic nie robi bo i tak tego spring nie wykorzystuje i tak jak mówisz na każdy request wołana jest nowa instacja tej i innych klas